### DISCUSSION TITLE
Only a few Dungeons can pass the challenge properly

### DISCUSSION THREAD

**Discussion by lichleo** [Melledy/Grasscutter#1353] [Q&A]
Only a few Dungeons such as Midsummer Courtyard can pass the challenge properly and get the rewards.
Others may have the following problem:
1) After kill all monsters, no battle pass and the air wall is still there, so cannot go to get the rewards.
2) After kill group 1 monsters, second group monsters don't appear.

Is the lua script problem in resource file or some code logic need to be updated in the future?

**Comment by ytfh44** [comment]
yes，it is still too buggy
maybe u can look through the gc-resources
repo and contribute
